moduleroot/.github/workflows/publish.yaml.erb

```diff
@@ -0,0 +1,30 @@
+<% if configs['enabled'] -%>
+name: Build and publish to Puppet Forge
+
+on:
+ push:
+  tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Get latest tag
+      run: echo ::set-env name=RELEASE_VERSION::$(echo ${GITHUB_REF:10})
```

Comment:
If the checkout action does actually need the ref, then you could use github actions `set-output` instead like I do in (vscode)[https://github.com/puppetlabs/puppet-vscode/blob/1ec3cb73d3a6cf1675c35cb1e91fdf15ba2fa23c/.github/workflows/release.yml#L37] . That creates a variable that you can reference inside the [yaml](https://github.com/puppetlabs/puppet-vscode/blob/1ec3cb73d3a6cf1675c35cb1e91fdf15ba2fa23c/.github/workflows/release.yml#L57) instead of relying environment variables.

This might not be an improvement per say, and instead might be yak shaving. Also, beware I found `set-output` only works in bash.